test/unit/event.js

```diff
@@ -2552,34 +2552,69 @@ QUnit.test( "focusin using non-element targets", function( assert ) {
 testIframe(
 	"focusin from an iframe",
 	"event/focusinCrossFrame.html",
-	function( assert, framejQuery, frameWin, frameDoc ) {
+function( assert, framejQuery, frameWin, frameDoc ) {
 		assert.expect( 1 );
 
-		var input = jQuery( frameDoc ).find( "#frame-input" );
+		var done = assert.async(),
+			focus = false,
+			input = jQuery( frameDoc ).find( "#frame-input" );
 
 		// Create a focusin handler on the parent; shouldn't affect the iframe's fate
 		jQuery( "body" ).on( "focusin.iframeTest", function() {
 			assert.ok( false, "fired a focusin event in the parent document" );
 		} );
 
 		input.on( "focusin", function() {
+			focus = true;
 			assert.ok( true, "fired a focusin event in the iframe" );
 		} );
 
 		// Avoid a native event; Chrome can't force focus to another frame
-		input.trigger( "focusin" );
-
-		// Must manually remove handler to avoid leaks in our data store
-		input.remove();
-
-		// Be sure it was removed; nothing should happen
-		input.trigger( "focusin" );
+		input[ 0 ].focus();
 
 		// Remove body handler manually since it's outside the fixture
 		jQuery( "body" ).off( "focusin.iframeTest" );
+
+		setTimeout( function() {
+
+			// DOM focus is unreliable in TestSwarm
+			if ( QUnit.isSwarm && !focus ) {
+				assert.ok( true, "GAP: Could not observe focus change" );
```

Comment:
@gibson042 BTW, I copied this from other similar assertions but... what is GAP?

Reply:
My attempt to make clear that this is not a true passing assertion, but rather a failure to verify the desired assertion at all... a **gap** between intent and capability.

Reply:
Ah. Since it’s all uppercase, I thought it’s an acronym. 😅